### ISSUE TITLE
Remove references to chrome-infra-auth from luci-go

### ISSUE THREAD

**Issue by maruel** [luci/luci-go#25]
We need to make the following a flag:

    common/auth/service.go:const ServiceURL string = "https://chrome-infra-auth.appspot.com"

Right now it is impossible to authenticate with another auth_service instance.

**Comment by adrianludwin** [comment]
So it turns out that the hardcoded ServiceURL is a) easy to override, even though this hasn't yet been done and b) not particularly important, as it's only used to display a bit of info from the authutil CLI.

The current state is actually that by using "authutil login", you end up authorization the *client*, which can then talk to *any* luci instance on your behalf. That is, I logged in using authutil, then pointed my client at foundry-isolate.appspot.com and it worked just fine.

So - all done? Not quite. Firstly, when I visited the URL produced by authutil, it told me that I was logging into Chrome Infrastructure, based on the OAuth ClientID and client secret that's hardcoded into luci-go. Ideally, I should be able to point authutil to (for example) myfoo-authorization.appspot.com, and as a result have my OAuth provider tell me that I'm logging into "MyFoo" and not Chrome Infra.

This then implies that my authorizations should be kept separate depending on which server I'm talking to - so if I logged into MyFoo, I would have to log in again for Chrome, and luci-go would keep this all straight for me. 

Any luci installation has  multiple base URLs, but I'd only want to log into the MyFoo "project" once - this could be accomplished by allowing any MyFoo service (eg myfoo-isolate.appspot.com) redirect the login request to myfoo-authorization.appspot.com, allowing the client to keep track of a single canonical URL per project.

**Comment by vadimsht** [comment]
You are correct, ServiceURL is not essential. I was planning to remove it (and use Google's `/tokeninfo?access_token` endpoint in `authutil info` to get user info).

> The current state is actually that by using "authutil login", you end up authorization the client, which can then talk to any luci instance on your behalf. That is, I logged in using authutil, then pointed my client at myfoo-isolate.appspot.com and it worked just fine.

This would work only if OAuth ClientID is explicitly whitelisted in myfoo-isolate.appspot.com config (see /auth/oauth_config).

> Any luci installation has multiple base URLs, but I'd only want to log into the MyFoo "project" once - this could be accomplished by allowing any MyFoo service (eg myfoo-isolate.appspot.com) redirect the login request to myfoo-authorization.appspot.com, allowing the client to keep track of a single canonical URL per project.

This is exactly how predecessor of authutil works: https://github.com/luci/luci-py/blob/master/client/utils/oauth.py#L225 (For some base URL, it fetches `<baseURL>/auth/api/v1/server/oauth_config` that describes URL of central Auth Service, as well as ClientID and Client (not so) Secret. Auth Service URL is used as a key in local token cache).

I felt it's too complicated for `authutil` and I wanted to support a case of "login once, everywhere" (that OAuth essentially implements, since an oauth token is universally usable).

---

I don't have the context here. What problem are we trying to solve? I think I can add "per installation" mode, if needed.

**Comment by vadimsht** [comment]
(Or alternatively make OAuth clientID and secret configurable, per binary, similar to proposal here https://github.com/luci/luci-go/issues/26#issuecomment-240518061)

**Comment by adrianludwin** [comment]
I felt it was a bit weird to be authorizing the client to act on your behalf with _any_ luci stack, even if (as you said) your OAuth was explicitly whitelisted in those stacks. However, maybe it's ok - but then we'll need to change the luci client ID from "Chrome Infrastructure" to something more generic, IMO. I'll ask my team what they think too.
